core/src/main/java/io/grpc/ServerProvider.java

```diff
@@ -35,6 +35,13 @@
   @VisibleForTesting
   static final ServerProvider load(ClassLoader cl) {
     ServiceLoader<ServerProvider> providers = ServiceLoader.load(ServerProvider.class, cl);
+
+    // Attempt to load using the context class loader and ServiceLoader.
+    // This allows frameworks like http://aries.apache.org/modules/spi-fly.html to plug in.
+    if (!providers.iterator().hasNext() && cl.equals(ServerProvider.class.getClassLoader())) {
```

Comment:
Instead of doing the equals hack in the test, how about we just drop this part of the condition. Let's drop it in all the providers? If it makes the tests too painful then maybe we just pass in a `boolean fallbackToContextCl` for the test to simplify the behavior.

Reply:
Actually i don't think it is needed for the tests at all. I will drop it.